modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TestflowForm.tsx

```diff
@@ -222,6 +224,23 @@ const TestflowForm = () => {
 		setValue('userIds', userIds);
 	}, [setValue, userIds]);
 
+	useEffect(() => {
+		if (caseTypesWatch.length && caseTypesWatch.length < caseTypes.length) {
+			setCheckedAll(false);
+		}
+	}, [caseTypes.length, caseTypesWatch.length]);
+
+	const handIsAllChecked = () => {
+		if (!isCheckedAll) {
+			caseTypes.map((caseType, index) => {
+				setValue(`caseTypes.${index}`, caseType.id);
+			});
+		}
+		else {
+			setValue('caseTypes', []);
+		}
+	};
```

Comment:
1. Reverse the order
2. Change the name (we use `on`) as name convection for functions
3. forEach over map, since forEach returns nothing

```suggestion
	const onSelectAll = () => {
		if (isCheckedAll) {
		        setValue('caseTypes', []);
		}
		else {
			caseTypes.forEach((caseType, index) => {
				setValue(`caseTypes.${index}`, caseType.id);
			});
		}
	};
```

Reply:
Done!